src/domain/model/user/Address.ts

```diff
@@ -1,12 +1,43 @@
 import { Country } from './Country';
+import { DomainValidationError } from '../DomainValidationError';
 
 export class Address {
   constructor(
     public readonly address1: string,
     readonly address2: string | undefined,
     readonly city: string,
-    readonly region: string,
+    readonly region: string | undefined,
```

Comment:
Shouldn't be this `null` instead of `undefined`?

Reply:
Typescript works better with undefined (when you specify an argument as `?` it marks it as optional which translates to (type | undefined) so it is more interoperable. Semantically it isn't that different because technically the customer really didn't define a region